### ISSUE TITLE
Loss Convergence Benchmark

### ISSUE THREAD

**Issue by anijain2305** [pytorch/torchdynamo#713]
This point has been raised multiple times. Our benchmarks today only focus on single iteration of fwd-bwd (and optimizers soon). But we lack a real training example, with loss going down.

We could start with Bert model and run for say 500 iterations - with both pre-training and fine-tuning batch sizes. Nvidia folks mentioned that this could reveal many bugs which are not visible in single iteration.

**Comment by yanboliang** [comment]
Update:
* [x] Start with a huggingface Bert model and set up the end to end train/evaluation workflow with basic configs. 
* [ ] Enable TorchDynamo for model training and evaluation function, and ensure the loss going down.
  - [ ] ```torch.ops.profiler._record_function_exit``` returns a tensor to store underlying ```RecordFunction```, but dynamo can't handle this. Current we put this function into FX graph, but doesn't set its return value as one of the graph output. 
  - [ ] ```WithExitFunctionVariable``` can't reconstruct ```ProfileRecordFunctionVariable``` correctly.
  - [ ] ```ProfileRecordFunctionVariable``` misses ```with``` context on the graph break instruction. I think we should refer [how ```GradModeVariable``` handling this case](https://github.com/pytorch/torchdynamo/issues/207).
